apps/dsb-client-gateway-api/src/app/modules/dsb-client/service/dsb-api.service.ts

```diff
@@ -179,22 +189,33 @@ export class DsbApiService implements OnModuleInit {
     return data;
   }
 
-  public async getTopics(ownerDID?: string): Promise<TopicDataResponse> {
-    if (!ownerDID) {
+  public async getApplicationsByOwnerAndRole(
+    roleName: string
+  ): Promise<ApplicationDTO[]> {
+    try {
       const enrolment = this.enrolmentRepository.getEnrolment();
+      const ownerDID = enrolment.did;
+      return this.iamService.getApplicationsByOwnerAndRole(roleName, ownerDID);
+    } catch (e) {
+      this.logger.error(e);
```

Comment:
```suggestion
      this.logger.error('error while getting applications', e);
```

It will be much easier to identify the source of error later